lib/Form/Checkbox/Group.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Checkbox from './';
+
+export default class Group extends Component {
+  static propTypes = {
+    choices: PropTypes.arrayOf(PropTypes.shape({
+
+    })).isRequired,
+    onChangeHandler: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const selected = props.choices.filter(c => c.checked);
+    this.state = { choices: props.choices, selected };
+    this.onChangeHandler = this.onChangeHandler.bind(this);
+    this.choiceIsAlreadySelected = this.choiceIsAlreadySelected.bind(this);
+    this.removeChoice = this.removeChoice.bind(this);
+    this.addChoice = this.addChoice.bind(this);
+  }
+
+  onChangeHandler(e) {
+    const choice = this.state.choices.find(c => c.id === e.target.id);
+
+    const selected = this.choiceIsAlreadySelected(choice) ? this.removeChoice(choice) : this.addChoice(choice);
+    this.setState({ selected });
+
+    this.props.onChangeHandler(selected);
+  }
+
+  choiceIsAlreadySelected(choice) {
+    return this.state.selected.findIndex(c => c === choice) >= 0;
+  }
+
+  addChoice(choice) {
+    return this.state.selected.concat(choice);
```

Comment:
State should only be manipulated via setState, here's some docs to explain why;

- https://facebook.github.io/react/docs/react-component.html#setstate
- https://stackoverflow.com/questions/37755997/why-cant-i-directly-modify-a-components-state-really

Reply:
With this in mind you could then pass `this.props.onChangeHandler(selected);` as a callback to the `setState` method chaining them nicely together. e.g.

```
addChoice(choice) {
  return this.setState({
    selected: this.state.selected.concat({
      ...choice,
      selected: true
    })
  }, this.onChangeHandler);
}

onChangeHandler() {
  this.props.onChangeHandler(this.state.selected);
}
```

Reply:
The idea is shared with `removeChoice`.